packages/site/cms-site/src/iframebridge/IFrameBridge.tsx

```diff
@@ -85,38 +86,47 @@ export const IFrameBridgeProvider = ({ children }: PropsWithChildren) => {
+        const newPreviewElementsData = previewElements
+            .map((previewElement) => {
+                const childNodes = getRecursiveChildrenOfPreviewElement(previewElement.element);
+                const positioning = getCombinedPositioningOfElements(childNodes);
 
-                    return {
-                        adminRoute: previewElement.adminRoute,
-                        label: previewElement.label,
-                        position: {
-                            top: positioning.top,
-                            left: positioning.left,
-                            width: positioning.width,
-                            height: positioning.height,
-                        },
-                    };
-                })
-                .sort((previousElementData, nextElementData) => {
-                    const previousSize = previousElementData.position.width * previousElementData.position.height;
-                    const nextSize = nextElementData.position.width * nextElementData.position.height;
-                    return nextSize - previousSize;
-                })
-                .map((elementData, index) => {
-                    return {
-                        ...elementData,
-                        position: {
-                            ...elementData.position,
-                            zIndex: index + 1,
-                        },
-                    };
-                }),
-        );
+                return {
+                    adminRoute: previewElement.adminRoute,
+                    label: previewElement.label,
+                    position: {
+                        top: positioning.top,
+                        left: positioning.left,
+                        width: positioning.width,
+                        height: positioning.height,
+                    },
+                };
+            })
+            .sort((previousElementData, nextElementData) => {
+                const previousSize = previousElementData.position.width * previousElementData.position.height;
+                const nextSize = nextElementData.position.width * nextElementData.position.height;
+                return nextSize - previousSize;
+            })
+            .map((elementData, index) => {
+                return {
+                    ...elementData,
+                    position: {
+                        ...elementData.position,
+                        zIndex: index + 1,
+                    },
+                };
+            });
+
+        setPreviewElementsData((previousElementsData) => {
+            const dataDidNotChange = isEqual(previousElementsData, newPreviewElementsData);
+
+            if (dataDidNotChange) {
+                // Returning the previous object (same reference) prevents the state-update from triggering a re-render
+                return previousElementsData;
+            }
+
+            return newPreviewElementsData;
+        });
```

Comment:
@nsams @manuelblum what do you think about this? Is this a valid setState call?

Reply:
why do you think it might be invalid? Or do you see possible performance issues with the deep isEqual?